frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/FileDownloader.java

```diff
@@ -83,7 +88,28 @@ private CloseableHttpResponse execute(String requestUrl) throws IOException {
             return executeViaProxy(proxy, requestUrl);
         } else {
             LOGGER.info("No proxy was configured, downloading directly");
-            response = buildHttpClient(null).execute(new HttpGet(requestUrl));
+            if (userName != null && !"".equals(userName) && password != null && !"".equals(password)) {
+                LOGGER.info("Using credentials (" + userName + ") from settings.xml");
+                // Auth target host
+                URL aURL = new URL(requestUrl);
+                HttpHost target = new HttpHost (aURL.getHost(), aURL.getPort(), aURL.getProtocol());
+                // Create AuthCache instance
+                AuthCache authCache = new BasicAuthCache();
+                // Generate BASIC scheme object and add it to the local auth cache
+                BasicScheme basicAuth = new BasicScheme();
+                authCache.put(target, basicAuth);
+                // Add AuthCache to the execution context
+                HttpClientContext localContext = HttpClientContext.create();
+                localContext.setAuthCache(authCache);
+                CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+                credentialsProvider.setCredentials(
+                        new AuthScope(aURL.getHost(), aURL.getPort()),
+                        new UsernamePasswordCredentials(userName, password)
+                );
+                response = buildHttpClient(credentialsProvider).execute(new HttpGet(requestUrl),localContext);
```

Comment:
Maybe extract a method `makeCredentialsProvider(...)` here? And `makeLocalContext(...)`?